src/main/java/com/coreos/jetcd/resolver/URIResolvers.java

```diff
@@ -0,0 +1,130 @@
+package com.coreos.jetcd.resolver;
+
+import com.coreos.jetcd.exception.EtcdExceptionFactory;
+import com.google.auto.service.AutoService;
+import io.grpc.EquivalentAddressGroup;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.naming.NamingEnumeration;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+
+public final class URIResolvers {
+
+  private URIResolvers() {
+  }
+
+  @AutoService(URIResolver.class)
+  public static final class Direct implements URIResolver {
+    private ConcurrentMap<URI, List<EquivalentAddressGroup>> cache;
+
+    public Direct() {
+      this.cache = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public int priority() {
+      return Integer.MIN_VALUE;
+    }
+
+    @Override
+    public boolean supports(URI uri) {
+      return "http".equals(uri.getScheme()) || "https".equals(uri.getScheme());
+    }
```

Comment:
we can use checking code in `isValidEndpointFormat` for Direct resolver's `supports`.

e.g

```java
private static final List<String> DRIECT_SCHEMES = Arrays.asList("http", "https");

public boolean supports(URI uri) {
    if (SCHEMES.stream().noneMatch(u.getProtocol()::equals)) {		
      return false;		
    }
    
    // endpoint must contain a port.		
    if (u.getPort() == -1) {		
      return false;		
    }
    
    // endpoint must not contain a path.		
    return u.getPath().isEmpty();	
}
```